source/adapters/cuda/command_buffer.hpp

```diff
@@ -9,7 +9,233 @@
 //===----------------------------------------------------------------------===//
 
 #include <ur/ur.hpp>
+#include <ur_api.h>
 
-/// Stub implementation of command-buffers for CUDA
+#include "context.hpp"
+#include <cuda.h>
+#include <memory>
 
-struct ur_exp_command_buffer_handle_t_ {};
+static auto getUrResultString = [](ur_result_t Result) {
+  switch (Result) {
+  case UR_RESULT_SUCCESS:
+    return "UR_RESULT_SUCCESS";
+  case UR_RESULT_ERROR_INVALID_OPERATION:
```

Comment:
I'm not too familiar with X macros but from a little research it seems like it wouldn't reduce much verbosity here, and that to do so it would probably need to be defined along with the `ur_result_t` enum itself in common headers, which would reduce the need for this here. That sounds to me like a potentially useful change, though probably out of scope for this PR. 

Please let me know if I'm misunderstanding though.